### DISCUSSION TITLE
synaptic delay values different

### DISCUSSION THREAD

**Discussion by wagdy88** [jonescompneurolab/hnn-core#563] [Q&A] [answered]
Hello, 

I am working on extracting values for the synaptic delay for one of my codes. I noticed that the values have been inconsistent for the distal delay between net.connectivity and net._params 

In net._params, the distal delay is as follows:
"input_dist_A_delay_L2": 5.0,
"input_dist_A_delay_L5": 5.0,

However, in net.connectivity, the distal delay is as follows:
evdist1 -> L5_basket
 cell counts: 35 srcs, 35 targets
 connection probability: 1.0 
 loc: 'distal'; receptor: 'ampa'
 weight: 0.0; delay: 0.1; lamtha: 3.0
  ,
 evdist1 -> L5_pyramidal
 cell counts: 100 srcs, 100 targets
 connection probability: 1.0 
 loc: 'distal'; receptor: 'ampa'
 weight: 0.25807; delay: 0.1; lamtha: 3.0
  ,
 evdist1 -> L5_pyramidal
 cell counts: 100 srcs, 100 targets
 connection probability: 1.0 
 loc: 'distal'; receptor: 'nmda'
 weight: 0.156725; delay: 0.1; lamtha: 3.0

Which one is the correct one?

Also, how can I extract data from net.connectivity if it's the correct one?

Thanks

**Comment by rythorpe** [comment, marked as answer]
`net.connectivity` should be the correct one. `net._params` is a private attribute that is used internally, so I would discourage anyone from trying to interpret any values specified in this dictionary.

**Comment by jasmainak** [comment]
`net.connectivity` is a dictionary, so you just need to use the keys. E.g., `net.connectivity['evdist1']['loc']`

**Reply by wagdy88** [reply]
Hi @jasmainak,
I tried this and it returns a list not a dictionary. Here's what I get when I check the type:
type(net.connectivity)
Out[3]: list

Is there a way to get the values directly from it?
Thanks 
Mohamed

**Reply by rythorpe** [reply]
Oh yes, `net.connectivity` is a list of dict. I recommend taking a look at our [connectivity tutorial](https://jonescompneurolab.github.io/hnn-core/stable/auto_examples/howto/plot_connectivity.html#sphx-glr-auto-examples-howto-plot-connectivity-py) as it will explain how to set and view network connectivity.

**Comment by wagdy88** [comment]
Thank you!

**Comment by jasmainak** [comment]
@wagdy88 for usage questions, let us try to use the discussions page instead of the bug tracker: https://github.com/jonescompneurolab/hnn-core/discussions. I'm going to convert this issue into a discussion for now.
